### ISSUE TITLE
Record.get(Field) doesn't work for embeddable table fields when non field-replacing embeddables are projected explicitly

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#10828]
Calling

```java
Record r = ...
EmbeddedFieldRecord e = r.get(T.EMBEDDED_FIELD);
```

Doesn't seem to work, it produces `null`. See report by @etiennestuder here: https://github.com/jOOQ/jOOQ/issues/6124#issuecomment-719479843

**Comment by lukaseder** [comment]
This doesn't reproduce it:

```java
@Test
public void testEmbeddablesRecordAccessByField() {
    assertEquals(1, ctx.insertInto(A1)
        .columns(A1.AUDIT)
        .values(audit)
        .execute());

    A1Record r = ctx.fetchOne(A1);
    assertEquals(audit, r.get(A1.AUDIT));
    assertEquals(audit.getId(), r.get(A1.AUDIT).getId());
    assertEquals(audit.getDc(), r.get(A1.AUDIT).getDc());
    assertEquals(audit.getDm(), r.get(A1.AUDIT).getDm());
    assertEquals(audit.getUc(), r.get(A1.AUDIT).getUc());
    assertEquals(audit.getUm(), r.get(A1.AUDIT).getUm());

    assertEquals(audit, r.getAudit());
    assertEquals(audit.getId(), r.getId());
    assertEquals(audit.getDc(), r.getDc());
    assertEquals(audit.getDm(), r.getDm());
    assertEquals(audit.getUc(), r.getUc());
    assertEquals(audit.getUm(), r.getUm());
}
```

This does:

```java
@Test
public void testEmbeddablesRecordAccessByFieldFromProjection() {
    assertEquals(1, ctx.insertInto(A1)
        .columns(A1.AUDIT)
        .values(audit)
        .execute());

    Record r = ctx.select(A1.AUDIT).from(A1).fetchOne();
    assertEquals(audit, r.get(A1.AUDIT));
    assertEquals(audit.getId(), r.get(A1.AUDIT).getId());
    assertEquals(audit.getDc(), r.get(A1.AUDIT).getDc());
    assertEquals(audit.getDm(), r.get(A1.AUDIT).getDm());
    assertEquals(audit.getUc(), r.get(A1.AUDIT).getUc());
    assertEquals(audit.getUm(), r.get(A1.AUDIT).getUm());
}
```

**Comment by lukaseder** [comment]
The generated query is correct:

```sql
select "A1"."ID", "A1"."DC", "A1"."DM", "A1"."UC", "A1"."UM" from "A1"
```

The problem seems to be that the result set contains the embeddable, despite the replacesfields flag not being active:

```
+--------------------------------------------------+
|AUDIT                                             |
+--------------------------------------------------+
|(1, timestamp '2000-01-01 00:00:00.0', null, 'u...|
+--------------------------------------------------+
```

It should contain the actual columns, which it does in the first test case:

```
+----+---------------------+------+----+------+------+------+
|  ID|DC                   |DM    |UC  |UM    |     X|     Y|
+----+---------------------+------+----+------+------+------+
|   1|2000-01-01 00:00:00.0|{null}|user|{null}|{null}|{null}|
+----+---------------------+------+----+------+------+------+
```

**Comment by lukaseder** [comment]
Fixed for jOOQ 3.15.0 and 3.14.2 (#10861)

**Comment by etiennestuder** [comment]
Cool. Thanks for the quick fix, @lukaseder .
